lib/matplotlib/pyplot.py

```diff
@@ -542,6 +530,11 @@ def make_active(event):
 
         _pylab_helpers.Gcf.set_active(figManager)
         figManager.canvas.figure.number = num
+        if _INSTALL_FIG_OBSERVER:
+            def auto_draw(fig):
+                if fig.stale and matplotlib.is_interactive():
+                    fig.canvas.draw_idle()
+            figManager.canvas.figure.add_callback(auto_draw)
```

Comment:
should we record the callback id somewhere, maybe to make it possible to disable the auto_draw() if plt.ioff() is called?